readthedocs/rtd_tests/tests/projects/test_version_sorting.py

```diff
@@ -0,0 +1,67 @@
+from django.test import TestCase
+from django_dynamic_fixture import get
+
+from readthedocs.builds.constants import BRANCH
+from readthedocs.builds.models import Version
+from readthedocs.projects.models import Project
+from readthedocs.projects.templatetags.projects_tags import sort_version_aware
+
+
+class SortVersionsTest(TestCase):
+
+    def setUp(self):
+        self.project = get(Project)
+
+    def test_basic_sort(self):
+        identifiers = ['1.0', '2.0', '1.1', '1.9', '1.10']
+        for identifier in identifiers:
+            get(
+                Version,
+                project=self.project,
+                type=BRANCH,
+                identifier=identifier,
+                verbose_name=identifier,
+                slug=identifier,
+            )
+
+        versions = list(Version.objects.filter(project=self.project))
+        self.assertEqual(
+            ['latest', '2.0', '1.10', '1.9', '1.1', '1.0'],
+            [v.slug for v in sort_version_aware(versions)],
+        )
+
+    def test_sort_wildcard(self):
+        identifiers = ['1.0.x', '2.0.x', '1.1.x', '1.9.x', '1.10.x']
+        for identifier in identifiers:
+            get(
+                Version,
+                project=self.project,
+                type=BRANCH,
+                identifier=identifier,
+                verbose_name=identifier,
+                slug=identifier,
+            )
+
+        versions = list(Version.objects.filter(project=self.project))
+        self.assertEqual(
+            ['latest', '2.0.x', '1.10.x', '1.9.x', '1.1.x', '1.0.x'],
+            [v.slug for v in sort_version_aware(versions)],
+        )
+
+    def test_sort_alpha(self):
+        identifiers = ['banana', 'apple', 'carrot']
+        for identifier in identifiers:
+            get(
+                Version,
+                project=self.project,
+                type=BRANCH,
+                identifier=identifier,
+                verbose_name=identifier,
+                slug=identifier,
+            )
+
+        versions = list(Version.objects.filter(project=self.project))
+        self.assertEqual(
+            ['latest', 'carrot', 'banana', 'apple'],
```

Comment:
I know that this is not a behavior that you changed, but I'd like to sort alphanumeric versions from A to Z instead at some point. Feels more human-readable when you have a many of these.

Reply:
Do we have any examples of projects that use something like this? I can think of Ubuntu off the top of my head, but that's about it.

Reply:
test-builds, which may not apply as a real project but shows the sorting problem when building alphanumeric branch names